src/TabSwipe/widget/HammerCarousel.ts

```diff
@@ -0,0 +1,275 @@
+import * as Hammer from "hammerjs";
+import * as domStyle from "dojo/dom-style";
+import { TabContainer, TabPan } from "./TabSwipe";
+
+export interface SwipeOptions {
+    tabContainer: TabContainer;
+    contentContainer: HTMLElement;
+    effect: "moveIn" | "moveOver";
+}
+
+export class HammerSwipe {
+    private container: HTMLElement;
+    private options: SwipeOptions;
+    private tabPanes: TabPan[];
+    private visibleTabs: TabPan[];
+    private activePan: TabPan;
+
+    public panes: HTMLElement[];
+    private containerSize: number;
+    private currentIndex: number;
+    private hammer: HammerManager;
+    private threshold: number;
+    private resizeTimer: number | null;
+    private ticking: boolean;
+
+    constructor(options: SwipeOptions) {
+        this.options = options;
+        this.threshold = 20;
+        this.currentIndex = 0;
+        this.ticking = false;
+        this.container = options.contentContainer;
+        this.panes = [].slice.call(this.container.children);
+        this.tabPanes = [].slice.call(options.tabContainer.getChildren());
+        this.activePan = options.tabContainer._active;
+
+        this.containerSize = this.container.offsetWidth;
+        this.hammer = new Hammer.Manager(this.container);
+        this.hammer.add(new Hammer.Pan({ threshold: 10 }));
+        this.hammer.on("panstart", event => this.onPanStart(event));
+        this.hammer.on("panmove", event => this.onPanMove(event));
+        this.hammer.on("panend pancancel", event => this.onPanEnd(event));
+        this.hammer.on("swipeup swipedown", event => this.onPan(event)); // dummy useless
+        this.options.tabContainer.showTab(this.activePan);
+        this.showTab(this.activePan);
+        // this.show(this.currentIndex);
+    }
+    /**
+     * Trottheled resize, setting calulating container size.
+     * resize should also be called when rotating the screen.
+     * But wait for the animation to finish, with a timeout.
+     */
+     resize() {
+        const resizeTimeout = 10000; // wait for animation to finish
+        if (this.resizeTimer) {
+            clearTimeout(this.resizeTimer);
+        }
+        this.resizeTimer = setTimeout(() => {
+            logger.debug("HammerCarousel.resize");
+            this.containerSize = this.container.offsetWidth;
+            this.show(this.currentIndex); // reselect to set proper sizing panels.
+        }, resizeTimeout);
+    }
+
+    showTab(seeTabPan: TabPan) {
+        this.visibleTabs = this.tabPanes.filter(tabPan => !tabPan._hidden);
+        this.visibleTabs.forEach((tabPan, index) => {
+            if (tabPan.index === seeTabPan.index) {
+                seeTabPan.visibilityIndex = index;
+                tabPan.visibilityIndex = index;
+                return;
+            }
+        });
+        this.visibleTabs.forEach((tabPan, index) => {
+            const diff = index - seeTabPan.visibilityIndex;
+            const transform = `translate3d(${diff * 100}%, 0, 0)`;
```

Comment:
For now, yes.
The alternative is to have the tab-container-content width = 100% *noOfVisibleTabs. probably we'll do that in next release